Automattic/wp-calypso
# CPT / State: Add restorePost action creator, refactor queries reducer

Related: #6434

This pull request seeks to add a new `restorePost` action creator. In implementing the reducer behaviors for these action types, I found that there was a common trend of "bail if queries not tracked for site, bail if receiving posts results in same instance, otherwise merge into state". So as not to worsen this problem, this pull request also refactors the `state.posts.queries` reducer, abstracting a common `applyToManager` function which calls a method with arguments on the manager for a site given the pattern described above.

__Testing instructions:__

`restorePost` is not yet used in the application (blocked by #6434).

Ensure that Mocha tests pass:

```
npm run test-client
```

Verify that no regressions occur in usage of post queryable data, particularly on [custom post types list screen](http://calypso.localhost:3000/types/post) (including filtering, trashing), or on the [post editor](http://calypso.localhost:3000/post) "Add Link" link to existing content (including search).

Test live: https://calypso.live/?branch=add/state-restore-post
